tests/unit/commands/test_init.py

```diff
@@ -100,3 +100,22 @@ def test_init_default(profile, name, project_dir, emitter, valid_new_dir, mocker
         "for reference information about the snapcraft.yaml format."
     )
     emitter.assert_message("Successfully initialised project.")
+
+
+@pytest.mark.parametrize("yaml_content", ["", "just a string"])
+def test_init_existing_yaml_invalid(yaml_content, emitter, valid_new_dir, mocker):
+    """Test the 'snapcraft init' command with existing empty/invalid snapcraft.yaml."""
+    snapcraft_yaml = pathlib.Path(valid_new_dir) / "snap/snapcraft.yaml"
+    cmd = _create_command(profile=None, project_dir=None, name=None)
+    mocker.patch.object(sys, "argv", cmd)
+
+    # Write snapcraft.yaml content
+    snapcraft_yaml.parent.mkdir()
+    with open(snapcraft_yaml, "w") as f:
+        f.write(yaml_content)
```

Comment:
```suggestion
    snapcraft_yaml.parent.mkdir(exist_ok=True)
    snapcraft_yaml.write_text(yaml_content)
```
Just a minor nitpick :) writing files is much nicer with pathlib